index.js

```diff
@@ -114,12 +122,16 @@ function compression (options) {
       }
 
       if (!stream) {
-        return _end.call(this, chunk, encoding)
+        return endOnce.call(this, chunk, encoding)
       }
 
       // mark ended
       ended = true
 
+      if (this.writableFinished || !this.socket.writable) {
```

Comment:
I believe that the essential condition here is `!this.socket.writable` but checking `this.writableFinished` seems reasonable as well.